src/js/components/ConnectionModals/ConnectionForm.tsx

```diff
@@ -127,7 +126,8 @@ const ConnectionForm = ({onClose, conn}: Props) => {
         return obj
       }, {})
       try {
-        await dispatch(initConnection(toCluster({host, name})))
+        const id = conn && conn.id
+        await dispatch(initConnection(toCluster({id, host, name}), true))
```

Comment:
Was this where the bug you were referring to was?

Reply:
Ah shoot, sorry I never saved my inline comments. Yes this is one of them. The problem here is that since I've started using generated id's for non-default connections, whenever we try to create a new connection and it fails we end up polluting connectionStatuses. As I say it out loud it occurs to me that we won't be persisting that state anyway though, so perhaps I will just remove this and let the state clean itself up on app reload